lib/src/metta/runner/stdlib.rs

```diff
@@ -1160,6 +1160,40 @@ impl CustomExecute for IntersectionAtomOp {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct IndexAtomOp {}
+
+grounded_op!(IndexAtomOp, "index-atom");
+
+impl Grounded for IndexAtomOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_EXPRESSION, ATOM_TYPE_ATOM, ATOM_TYPE_ATOM])
```

Comment:
I would use `Number` as an index type like in all arithmetic operations.
```suggestion
        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_EXPRESSION, ATOM_TYPE_NUMBER, ATOM_TYPE_ATOM])
```